splinterd/src/config/partial.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(not(feature = "config-toml"))]
+use std::fs::File;
+#[cfg(not(feature = "config-toml"))]
+use std::io::Read;
+
+#[cfg(not(feature = "config-toml"))]
+use toml;
+
+#[cfg(not(feature = "config-toml"))]
+use crate::config::ConfigError;
+
+/// PartialConfig is an intermediate representation of configuration values, used when combining
+/// several sources. As such, all values of the PartialConfig are options as it is not necessary
+/// to provide all values from a single source.
+#[derive(Deserialize, Default, Debug)]
+pub struct PartialConfig {
+    storage: Option<String>,
+    transport: Option<String>,
+    cert_dir: Option<String>,
+    ca_certs: Option<String>,
+    client_cert: Option<String>,
+    client_key: Option<String>,
+    server_cert: Option<String>,
+    server_key: Option<String>,
+    service_endpoint: Option<String>,
+    network_endpoint: Option<String>,
+    peers: Option<Vec<String>>,
+    node_id: Option<String>,
+    bind: Option<String>,
+    #[cfg(feature = "database")]
+    database: Option<String>,
+    registry_backend: Option<String>,
+    registry_file: Option<String>,
+    heartbeat_interval: Option<u64>,
+}
+
+impl PartialConfig {
+    /// Creates a new PartialConfig object from a toml file. Available to use when the `config-toml`
+    /// feature flag is not in use.
+    #[cfg(not(feature = "config-toml"))]
```

Comment:
This should be in a toml module that implements the TomlPartialConfigBuilder, not part of PartialConfig (PartialConfig shouldn't know anything about implementations).